### ISSUE TITLE
update version of IDEA inspection engine Version:2023.3.4 

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#14448]
we have stable failure of inspection validation 
https://app.circleci.com/pipelines/github/checkstyle/checkstyle/23624/workflows/e04a197a-8172-4262-9fa3-de02579854c4/jobs/506740/artifacts

[inspection-results/YAMLSchemaValidation.xml](https://output.circle-artifacts.com/output/job/ab162cce-e22e-4921-9f14-8dc473901855/artifacts/0/home/circleci/project/target/inspection-results/YAMLSchemaValidation.xml)
```
<problems is_local_tool="true">
<problem>
<file>file://$PROJECT_DIR$/.github/workflows/site.yml</file>
<line>14</line>
<module>project</module>
<entry_point TYPE="file" FQNAME="file://$PROJECT_DIR$/.github/workflows/site.yml"/>
<problem_class id="YAMLSchemaValidation" severity="WARNING" 
      attribute_key="WARNING_ATTRIBUTES">Validation by JSON Schema</problem_class>
<description>Schema validation: Property 'name' is not allowed</description>
<highlighted_element>name: "Site"</highlighted_element>
<language>yaml</language>
<offset>0</offset>
<length>12</length>
</problem>
```

such problems are false positives, that we do not know how to avoid. It use to be rare to have such violations, but now problem is stable.

TODO: update version of IDEA inspection engine with hope that it resolve this issue AND we do need to catch up on latest inspection version.
latest version is Version: 2023.3.3
https://www.jetbrains.com/idea/download/?section=linux

version to bump is at https://github.com/checkstyle/contribution/blob/master/devops/docker/idea-image/Dockerfile

There will be new violations: we need to suppress all of them and create issue to resolve them in separate PRs. If violations are easy to fix, ok to fix in the same time.

**Comment by romani** [comment]
example of docker build: https://github.com/checkstyle/contribution/pull/668

**Comment by nrmancuso** [comment]
We need to keep this issue open until we actually update the image version in this repo and restore yaml schema validation: https://github.com/MANISH-K-07/checkstyle/blob/90dd5f0957b24cc929cafdbab1b92c40bc00f245/config/intellij-idea-inspections.xml#L4109

**Comment by nrmancuso** [comment]
@MANISH-K-07 your new image is live: https://hub.docker.com/layers/checkstyle/idea-docker/jdk11-idea2023.3.4/images/sha256-eefc93eaebcc4d8e747115e539512220d6b41d437f93f2496eda27612b257f18?context=explore

```
➜  idea-image git:(master) ✗ git log -1 --oneline
80a6139 (HEAD -> master, upstream/master) Issue #14448: Updated IDEA version to v2023.3.4

➜  idea-image git:(master) ✗ docker build -t checkstyle/idea-docker:latest -t checkstyle/idea-docker:jdk11-idea2023.3.4 .
[+] Building 43.8s (9/9) FINISHED                                                                docker:default
 => [internal] load build definition from Dockerfile                                                       0.0s
 => => transferring dockerfile: 570B                                                                       0.0s
 => [internal] load metadata for docker.io/cimg/openjdk:11.0.16                                            0.0s
 => [internal] load .dockerignore                                                                          0.0s
 => => transferring context: 2B                                                                            0.0s
 => [1/5] FROM docker.io/cimg/openjdk:11.0.16                                                              0.1s
 => [2/5] RUN sudo apt update && sudo apt install -y wget                                                  4.4s
 => [3/5] RUN mkdir -p /home/circleci/idea                                                                 0.2s
 => [4/5] WORKDIR /home/circleci/idea                                                                      0.0s 
 => [5/5] RUN wget -qO- https://download.jetbrains.com/idea/ideaIC-2023.3.4.tar.gz |     tar xvz -C /hom  30.7s 
 => exporting to image                                                                                     8.2s 
 => => exporting layers                                                                                    8.2s 
 => => writing image sha256:6d395b16fbdea3e925f8b787ad657b75be58fc2a8647d351a88edbb3c4007c86               0.0s
 => => naming to docker.io/checkstyle/idea-docker:latest                                                   0.0s
 => => naming to docker.io/checkstyle/idea-docker:jdk11-idea2023.3.4                                       0.0s

➜  idea-image git:(master) ✗ docker push checkstyle/idea-docker:jdk11-idea2023.3.4
The push refers to repository [docker.io/checkstyle/idea-docker]
bb3b050bf0dd: Preparing 
5f70bf18a086: Preparing 
bb3b050bf0dd: Pushed 
057676229ef8: Pushed 
3027c90de795: Layer already exists 
36352a358ddc: Layer already exists 
2b7885936df5: Layer already exists 
b1c463e19a93: Layer already exists 
bdb5aba3e49e: Layer already exists 
ae2d1514e652: Layer already exists 
ebdabf56a236: Layer already exists 
73338609a9e4: Layer already exists 
bf659eac88cb: Layer already exists 
6d1a65ee9fdf: Layer already exists 
c3f11d77a5de: Layer already exists 
jdk11-idea2023.3.4: digest: sha256:eefc93eaebcc4d8e747115e539512220d6b41d437f93f2496eda27612b257f18 size: 3486
```

**Comment by nrmancuso** [comment]
> Great !!
> 
> Please let me know when I can update the image in the draft PR #14604

Let’s do it, we will also need to suppress new violations to make CI happy.

**Comment by MANISH-K-07** [comment]
@nrmancuso , @romani , image updated at #14696  :)
